waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -923,7 +933,29 @@
            :resources {:limits {:memory memory}
                        :requests {:cpu cpu :memory memory}}
            :volumeMounts [{:mountPath "/srv/www"
-                           :name "user-home"}]})))))
+                           :name "user-home"}]}))
+
+      ;; Optional envoy sidecar container
+      has-reverse-proxy?
+      (update-in
+        [:spec :template :spec :containers]
+        conj
+        (let [{:keys [cmd resources image]} reverse-proxy
+              env (:env service-description)]
+             {:command cmd
+              :env (into [{:name "SERVICE_PORT" :value (str service-port)}
+                          {:name "PORT0" :value (str port0)}]
+                         (concat
+                           (for [[k v] base-env]
+                                {:name k :value v})
+                           (for [[k v] env]
+                                {:name k :value v})))
```

Comment:
Nitpick: We should merge the maps into a single environment map and loop through it to avoid conflicting values (esp. user overriding SERVICE_PORT) from getting through and leading to an unexpected configuration.

Reply:
Done